generated/nidcpower/nidcpower/session.py

```diff
@@ -4694,6 +4694,43 @@ def export_attribute_configuration_file(self, file_path):
         errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
         return
 
+    @ivi_synchronized
+    def create_advanced_sequence(self, sequence_name, sequence, set_as_active_sequence=True):
+        '''create_advanced_sequence
+
+        Test.
+
+        Note: Test.
+
+        Args:
+            sequence_name (str): Name of sequence.
+
+            sequence (list of int): Sequence.
+
+            set_as_active_sequence (bool): Specifies that this current sequence is active.
+
+        '''
+        # First we need to get all possible properties we might be setting. The way the NI-DCPower C API is designed,
+        # we need to know this upfront in order to call `niDCPower_CreateAdvancedSequence`. In order to find the attribute
+        # ID of each property, we look at the member Attribute objects of Session.
+        attribute_ids_used = set()
+        for step in sequence:
+            for key in step:
+                if key not in Session.__base__.__dict__:
+                    raise TypeError('{0} is not an property on the nidcpower Session class'.format(key))
+                if not isinstance(Session.__base__.__dict__[key], _attributes.Attribute) and not isinstance(Session.__base__.__dict__[key], _attributes.AttributeEnum):
+                    raise TypeError('{0} is not an attribute type: {1}'.format(key, type(Session.__base__.__dict__[key])))
+                attribute_ids_used.add(Session.__base__.__dict__[key]._attribute_id)
```

Comment:
I think it's more Pythonic to simply try/except getting the _attribute_id rather than to use `isinstance`.

Reply:
But then we don't know the specific reason for the exception and cannot set the description appropriately.

Reply:
ok